src/Authenticator/Result.php

```diff
@@ -44,25 +44,28 @@ class Result implements ResultInterface
     protected $_errors = [];
 
     /**
-     * Sets the result code, identity, and failure messages
+     * Sets the result status, identity, and failure messages
      *
      * @param null|array|\ArrayAccess $data The identity data
-     * @param int $code Error code.
+     * @param string $status Status constant equivalent.
      * @param array $messages Messages.
      * @throws InvalidArgumentException When invalid identity data is passed.
      */
-    public function __construct($data, $code, array $messages = [])
+    public function __construct($data, $status, array $messages = [])
     {
-        if (empty($data) && $code === self::SUCCESS) {
+        if ($status === self::SUCCESS && empty($data)) {
```

Comment:
The boolean check is faster and if false the whole statement directly returns false afaik.

Reply:
faster is irrelevant in the scope of this functionality :) Especially on that level.

Reply:
Well it is also faster in my brain ;-) parsing empty() in my brain is probably the slowest operation existing ;-)